### PR TITLE
Fix when "Start a Review" button is enabled

### PR DESCRIPTION
**Pull request by StanleyGoldman**
Fixes: #1806 

Users cannot start a review from a comment if there are already comments in the thread.

- Correct the observable that is referenced to enable the `StartReview` command.
   https://github.com/github/VisualStudio/blob/8c9f77afaafd26d4bd2a641b46fff3b2c4b9958f/src/GitHub.InlineReviews/ViewModels/PullRequestReviewCommentViewModel.cs#L83-L85

` Added a count of comments in thread of `None` state into the criteria for `CanStartReview`
  - And used that information in determining if a review can be started.
   https://github.com/github/VisualStudio/blob/956d9d6f900e7f4831f88218887c0df3be79a816/src/GitHub.InlineReviews/ViewModels/PullRequestReviewCommentViewModel.cs#L61-L68

- Changed or removed tests that depending on being able to start a review after comments were made

### CODE REVIEW

**Comment by StanleyGoldman** [comment]
@grokys I could use your help understanding what some of these tests are intending to test and why some of them fail.

File: test/GitHub.InlineReviews.UnitTests/ViewModels/PullRequestReviewCommentViewModelTests.cs [hunk 1]
```diff
@@ -172,12 +191,12 @@ public void IsDisabledWhenSessionHasPendingReview()
...
 
             [Test]
-            public void IsDisabledWhenSessionHasNoPendingReview()
+            public void IsEnabledWhenSessionHasNoPendingReview()
             {
                 var session = CreateSession(false);
                 var target = CreateTarget(session);
 
-                Assert.That(target.StartReview.CanExecute(null), Is.False);
+                Assert.That(target.StartReview.CanExecute(null), Is.True);
```
Comment (grokys) [inline test/GitHub.InlineReviews.UnitTests/ViewModels/PullRequestReviewCommentViewModelTests.cs]: I don't think this is correct. This was asserting that you can't start a review when `Body` is empty. With this change the "Start review" button is enabled with no body:

image

The test should probably have been named `IsDisabledWhenSessionHasNoPendingReviewAndEmptyBody` to match the following `IsEnabledWhenSessionHasNoPendingReviewAndBodyNotEmpty`.

**Comment by grokys** [comment]
Hmm, actually is this correct?

image

The "Start a review" button appears for me for replies.

File: test/GitHub.InlineReviews.UnitTests/ViewModels/PullRequestReviewCommentViewModelTests.cs (hunk 1 above)
Comment (StanleyGoldman) [reply test/GitHub.InlineReviews.UnitTests/ViewModels/PullRequestReviewCommentViewModelTests.cs]: Okay cool, those tests make more sense now.

**Comment by StanleyGoldman** [comment]
There are two more tests that are failing for me that I don't immediately follow...

https://github.com/github/VisualStudio/blob/f0de5e86188d1694b4464d221616fdea437ef8e8/test/GitHub.InlineReviews.UnitTests/ViewModels/InlineCommentPeekViewModelTests.cs#L202-L236

https://github.com/github/VisualStudio/blob/f0de5e86188d1694b4464d221616fdea437ef8e8/test/GitHub.InlineReviews.UnitTests/ViewModels/InlineCommentPeekViewModelTests.cs#L238-L272

**Comment by StanleyGoldman** [comment]
I was just massively confused

**Pull request closed without merging**